Library/ViewModels/DiscoveryPageViewModel.swift

```diff
@@ -149,14 +149,16 @@ public final class DiscoveryPageViewModel: DiscoveryPageViewModelType, Discovery
 
     self.asyncReloadData = self.projects.take(first: 1).ignoreValues()
 
-    self.showEmptyState = paramsChanged
-      .takeWhen(paginatedProjects.filter { $0.isEmpty })
-      .map(emptyState(forParams:))
-      .skipNil()
-      .skipRepeats()
+    self.showEmptyState = Signal.combineLatest(paramsChanged, self.projectsAreLoading, paginatedProjects)
```

Comment:
`combineLatest` can be a tricky beast, just be sure it's what you want here!

Perhaps another way to do this to avoid the optional `EmptyState?` could be to use `filter`? Something like:
```
Signal.combineLatest(paramsChanged, self.projectsAreLoading, paginatedProjects)
  .filter { _, projectsAreLoading, projects in projectsAreLoading == false, projects.isEmpty else }
  .map { ... }
```

Reply:
Good suggestion, I don't know why I didn't think of this before!